allennlp/modules/token_embedders/bidirectional_language_model_token_embedder.py

```diff
@@ -0,0 +1,170 @@
+import json
+from typing import Dict, Tuple, TYPE_CHECKING
+
+import torch
+
+from allennlp.data import TokenIndexer, Token
+from allennlp.models.archival import load_archive
+from allennlp.modules.scalar_mix import ScalarMix
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.nn.util import remove_sentence_boundaries, get_text_field_mask, add_sentence_boundary_token_ids
+
+if TYPE_CHECKING:
+    from allennlp.models.bidirectional_lm import BidirectionalLanguageModel
+
+
+@TokenEmbedder.register('bidirectional_lm_token_embedder')
+class BidirectionalLanguageModelTokenEmbedder(TokenEmbedder):
+    """
+    Compute a single layer of representations from a bidirectional language model.
```

Comment:
Document here how these representations are computed (by fixing weights, and computing a learned scalar average of layers).